dos/src/main.rs

```diff
@@ -23,6 +29,53 @@ fn get_repair_contact(nodes: &[ContactInfo]) -> ContactInfo {
     contact
 }
 
+fn test_multisig_tx(
+    transaction_params: &TransactionParams,
+    rpc_client: &Option<RpcClient>,
+) -> Transaction {
+    let kpvals: Vec<Keypair> = (0..transaction_params.num_sign)
+        .map(|_| Keypair::new())
+        .collect();
+    let keypairs: Vec<&Keypair> = kpvals.iter().collect();
+
+    let blockhash = if transaction_params.valid_block_hash {
+        rpc_client.as_ref().unwrap().get_latest_blockhash().unwrap()
+    } else {
+        Hash::default()
+    };
+
+    let lamports = 5;
+    let transfer_instruction = SystemInstruction::Transfer { lamports };
+
+    let program_ids = vec![system_program::id(), stake::program::id()];
+
+    let instructions = vec![CompiledInstruction::new(
+        0,
+        &transfer_instruction,
+        vec![0, 1],
+    )];
+
+    let mut tx = Transaction::new_with_compiled_instructions(
+        &keypairs,
+        &[],
+        blockhash,
+        program_ids,
+        instructions,
+    );
+    if !transaction_params.valid_signatures {
+        tx.signatures = vec![Transaction::get_invalid_signature(); transaction_params.num_sign];
```

Comment:
signing is relatively expensive, in unsigned mode would probably just want to not sign instead of signing and then "un-signing"